kernel/pte_flags/src/pte_flags_aarch64.rs

```diff
@@ -0,0 +1,208 @@
+//! The aarch64-specific definitions of PTE flags.
+//!
+//! The definition of these flags assumed that the [MAIR] index 0
+//! has a "DEVICE nGnRE" entry, and [MAIR] index 1 has a Normal + Outer Shareable entry.
+//!
+//! [MAIR]: https://docs.rs/cortex-a/latest/cortex_a/registers/MAIR_EL1/index.html
+
+use crate::PteFlags;
+use bitflags::bitflags;
+use static_assertions::const_assert_eq;
+
+/// A mask for the bits of a page table entry that contain the physical frame address.
+pub const PTE_FRAME_MASK: u64 = 0x000_7FFFFFFFFF_000;
+
+// Ensure that we never expose reserved bits [12:50] as part of the ` interface.
+const_assert_eq!(PteFlagsAarch64::all().bits() & PTE_FRAME_MASK, 0);
+
+
+bitflags! {
+    /// Page table entry (PTE) flags on aarch64.
+    ///
+    /// **Note:** items beginning with an underscore `_` are not used in Theseus.
+    ///
+    /// The designation of bits in each `PageTableEntry` is as such:
+    /// * Bits `[0:11]` (inclusive) are reserved by hardware for access flags, cacheability flags,
+    ///   shareability flags, and TLB storage flags.
+    /// * Bits `[12:50]` (inclusive) are reserved by hardware to hold the physical frame address.
+    /// * Bits `[51:54]` (inclusive) are reserved by hardware for more access flags.
+    /// * Bits `[55:58]` (inclusive) are available for custom OS usage.
+    /// * Bits `[59:63]` (inclusive) are reserved by hardware for extended access flags.
```

Comment:
If we want to keep 48-bit OA mode, the correct documentation would be:
```rust
    /// * Bits `[0:11]` (inclusive) are reserved by hardware for access flags, cacheability flags,
    ///   shareability flags, and TLB storage flags.
    /// * Bits `[12:47]` (inclusive) are reserved by hardware to hold the physical frame address.
    /// * Bits `[48:49]` (inclusive) are ignored.
    /// * Bits `[50:54]` (inclusive) are reserved by hardware for more access flags.
    /// * Bits `[55:58]` (inclusive) are available for custom OS usage.
    /// * Bits `[59:63]` (inclusive) are reserved by hardware for extended access flags.
```

Reply:
Yes, let's stick with 48-bit OA size and also assume an equivalent 48-bit IPA size. Thanks for identifying this!